subprojects/dependency-management/src/main/java/org/gradle/internal/resolve/result/BuildableComponentResolveResult.java

```diff
@@ -40,4 +40,11 @@
      * Replaces the meta-data in the result. Result must already be resolved.
      */
     void setMetadata(ComponentResolveMetadata metadata);
+
+    /**
+     * Returns true if we should retry to get the resource when it was missing. This should only
+     * be used if the result of resolution shouldn't be impacted by a missing module (that is to say,
+     * it doesn't fail if a module is missing).
+     */
+    boolean isRetryMissing();
```

Comment:
This has the same issue as `BuildableModuleComponentMetaDataResolveResult.retryMissing`, in that it's adding a control flag to a type that is designed to be an output parameter. However, in this case I don't (yet) have a great alternative: adding it to `ComponentOverrideMetadata` isn't really any better.